coordinator/core/marbleapi.go

```diff
@@ -92,7 +92,7 @@ func (c *Core) Activate(ctx context.Context, req *rpc.ActivationReq) (*rpc.Activ
 	}
 
 	// Generate unique (= per marble) secrets
-	privateSecrets, err := c.data.getSecretMap(c.cmp.privateSecrets)
+	privateSecrets, err := c.data.getSecretMap(true)
```

Comment:
Can we simplify this by not distinguishing secrets in the store?
Instead, we could handle different secrets in `generateSecrets` and only generate the private ones.
This would simplify the code quite a bit.

Reply:
Since the `Shared` flag is stored with the secret it should be possible to generate the list at runtime by requesting the secrets and then checking whether or not they are private.
And since for a marble activation we need to request all shared secrets anyways there should be no performance difference.